### PR TITLE
Fixing go.mod snyk CVEs

### PR DESCRIPTION
**Pull request by DharmitD**
## The issue resolved by this Pull Request:
Resolves [RHOAIENG-2259](https://issues.redhat.com/browse/RHOAIENG-2259)

## Description of your changes:

Fixing go.mod dependencies/packages related snyk CVEs.
Imported my fork in synk and made sure the relevant CVEs were fixed with this update.

## Checklist
- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-562`

File: go.mod [hunk 1]
```diff
@@ -14,13 +14,13 @@ require (
...
 	github.com/prometheus/client_golang v1.12.2
 	github.com/spf13/viper v1.8.1
 	github.com/stretchr/testify v1.8.3
 	go.uber.org/zap v1.21.0
-	k8s.io/api v0.25.0
-	k8s.io/apimachinery v0.25.0
+	k8s.io/api v0.26.1
+	k8s.io/apimachinery v0.26.1
 	k8s.io/client-go v0.25.0
 	sigs.k8s.io/controller-runtime v0.13.0
```
Comment (HumairAK) [inline go.mod:25]: I think all k8s packages need to match in versions 

can you make sure whether these need to be updated as well

Comment (HumairAK) [reply go.mod:25]: Thanks @ChristianZaccaria!  

@DharmitD some references for you: 
* https://github.com/kubernetes/client-go?tab=readme-ov-file#versioning
* https://github.com/kubernetes-sigs/controller-runtime/blob/main/VERSIONING.md

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-562`

File: go.mod (hunk 1 above)
Comment (DharmitD) [reply go.mod]: Thanks @ChristianZaccaria , @HumairAK !
Done, updated  `client-go` to `v0.26.1` and `controller-runtime` to `0.14.7`. Confirmed that this version is compatible with the K8's packages' versions, i.e., `0.26.1` by inspecting the go.mod: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.14.7/go.mod#L20-L23

**Pull request merged**